pkg/skaffold/build/local/jib.go

```diff
@@ -18,16 +18,113 @@ package local
+	if err != nil {
+		return "", err
+	}
+	commandLine := append(maven, mavenCommand...)
+ 
+	err = executeBuildCommand(ctx, out, workspace, commandLine)
+	if err != nil {
+		return "", errors.Wrap(err, "maven build failed")
+	}
+	return skaffoldImage, nil
+}
+
+// generateMavenCommand generates the command-line to pass to maven for building a
+// project found in `workspace`.  The resulting image is added to the local docker daemon
+// and called `skaffoldImage`.
+func generateMavenCommand(workspace string, skaffoldImage string, a *v1alpha3.JibMavenArtifact) ([]string, error) {
+	if a.Module != "" {
+		// TODO: multi-module
+		return nil, errors.New("Maven multi-modules not supported yet")
+	}
+	// use mostly-qualified plugin ID in case jib is not a configured plugin
+	commandLine := []string{"prepare-package", "com.google.cloud.tools:jib-maven-plugin::dockerBuild", "-Dimage="+skaffoldImage}
+	if a.Profile != "" {
+		commandLine = append(commandLine, "-P"+a.Profile)
+	}
+	return commandLine, nil
 }
 
-func (b *Builder) buildJibGradle(_ /*ctx*/ context.Context, _ /*out*/ io.Writer, _ /*workspace*/ string, _ /*a*/ *v1alpha3.JibGradleArtifact) (string, error) {
-	return "", errors.New("buildJibGradle is unimplemented")
+func (b *Builder) buildJibGradle(ctx context.Context, out io.Writer, workspace string, a *v1alpha3.JibGradleArtifact) (string, error) {
+	skaffoldImage := generateJibImageRef(workspace, a.Project)
+	gradle, err := findBuilder("gradle", "gradlew", workspace)
+	if err != nil {
+		return "", err
+	}
+	gradleCommand, err := generateGradleCommand(workspace, skaffoldImage, a) 
+	if err != nil {
+		return "", err
+	}
+	commandLine := append(gradle, gradleCommand...)
+ 
+	err = executeBuildCommand(ctx, out, workspace, commandLine)
+	if err != nil {
+		return "", errors.Wrap(err, "gradle build failed")
+	}
+	return skaffoldImage, nil
+}
+
+// generateGradleCommand generates the command-line to pass to gradle for building an
+// project in `workspace`.  The resulting image is added to the local docker daemon
+// and called `skaffoldImage`.
+func generateGradleCommand(workspace string, skaffoldImage string, a *v1alpha3.JibGradleArtifact) ([]string, error) {
+	if a.Project != "" {
+		// multi-module
+		return []string{":" + a.Project + ":jibDockerBuild", "--image="+skaffoldImage}, nil
+	}
+	return []string{":jibDockerBuild", "--image="+skaffoldImage}, nil
+}
+
+// executeBuildCommand executes the command-line with the working directory set to `workspace`.
+func executeBuildCommand(ctx context.Context, out io.Writer, workspace string, commandLine []string) error {
+	logrus.Infof("Building %s: %s", workspace, commandLine)
+	cmd := exec.CommandContext(ctx, commandLine[0], commandLine[1:]...)
+	cmd.Dir = workspace
+	cmd.Stdout = out
+	cmd.Stderr = out
+	return util.RunCmd(cmd)
+}
+
+const (
+	// regexp matching valid image names
+	REPOSITORY_COMPONENT_REGEX string = `^[a-z\d]+(?:(?:[_.]|__|-+)[a-z\d]+)*$`
```

Comment:
Constants in Go are camelcase. `RepositoryComponentRegex`. Constants should also go in the `constants.go` package so they can be used multiple times without introducing cross package dependencies